client/rest/src/main/java/org/elasticsearch/client/RestClientActions.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.nio.protocol.HttpAsyncResponseConsumer;
+
+/**
+ * Actions that can be taken on a {@link RestClient} or the stateless views
+ * returned by {@link RestClientActions#withHostSelector withHostSelector}.
+ */
+public interface RestClientActions {
```

Comment:
I like having the interface because it is easier to reason about how things like Mockito will behave. I feel like we're more free to rework the implementations of things without breaking users if we have the interface too. And it is easier to reason about what is public.

Reply:
And I can stick all the javadocs here and they stay out of the way!

Reply:
ok